### ISSUE TITLE
lib.c plc_tag_read 

### ISSUE THREAD

**Issue by engerlf** [libplctag/libplctag#381]
Is the while loop condition correct at line number 1647？
while(rc == PLCTAG_STATUS_PENDING && time_ms() > end_time);
I think the correct spelling is
while(rc == PLCTAG_STATUS_PENDING && time_ms() < end_time);

**Comment by kyle-github** [comment]
Hmm, that does not look right at all.   Looking back in the local commits it looks like there was an erroneous merge done when I merged the original code into the prerelease branch.  

The only reason that there is a loop at all there is that in some cases the cond_wait() could return early.   I tried to guard against that in the implementation of cond_wait() but I did not trust that I had completely eliminated the possibility.

The same check is in plc_tag_write().

Thanks for finding this!
